### PR TITLE
virglrenderer: New Package

### PR DESCRIPTION
**Pull request by heljkon**
As requested in  #10144 

With minor modifications in https://github.com/kjliew/qemu-3dfx/blob/master/virgil3d/MINGW-packages/PKGBUILD

Tests won't compile because of missing iovec

### CODE REVIEW

**Comment by heljkon** [comment]
@lazka 
Just copied from https://github.com/kjliew/qemu-3dfx/blob/master/virgil3d/MINGW-packages/PKGBUILD 
Not tested!

Please review!

File: mingw-w64-virglrenderer/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+pkgver=0.9.1
+pkgrel=1
+pkgdesc='A virtual 3D GPU library, that allows the guest operating system to use the host GPU to accelerate 3D rendering'
+arch=('any')
+mingw_arch=('mingw64','mingw32','ucrt64','clang64','clang32')
+url='https://virgil3d.github.io/'
+license=(MIT)
+depends=("${MINGW_PACKAGE_PREFIX}-libepoxy")
+makedepends=("${MINGW_PACKAGE_PREFIX}-python"
+             "${MINGW_PACKAGE_PREFIX}-cmake"
```
Comment (lazka) [inline mingw-w64-virglrenderer/PKGBUILD:15]: Does it really need cmake?

**Comment by lazka** [comment]
lgtm

File: mingw-w64-virglrenderer/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+             "${MINGW_PACKAGE_PREFIX}-ninja")
+_tag=virglrenderer-$pkgver
+source=("virglrenderer-$pkgver.tar.bz2::https://gitlab.freedesktop.org/virgl/virglrenderer/-/archive/$_tag/virglrenderer-$_tag.tar.bz2" 
+	'https://raw.githubusercontent.com/kjliew/qemu-3dfx/7a8c114a910826cda99f06c6dfdfd438b1bedf31/virgil3d/MINGW-packages/0001-Virglrenderer-on-Windows-and-macOS.patch')
+sha256sums=('860cc3f0f1a448542a812b6e13909824eb3140b7b3283d068e9640487d50e6c0'
+            'ee129b1d13ae72b630c4f6c25501344dee0b6fb33fe5a0b03a86759d9666ded9')
+
+prepare() {
+  cd virglrenderer-$_tag
+  patch -p1 -i ${srcdir}/0001-Virglrenderer-on-Windows-and-macOS.patch
```
Comment (lazka) [inline mingw-w64-virglrenderer/PKGBUILD:26]:
```suggestion
  # https://gitlab.freedesktop.org/virgl/virglrenderer/-/issues/217
  patch -p1 -i ${srcdir}/0001-Virglrenderer-on-Windows-and-macOS.patch
```

Comment (lazka) [reply mingw-w64-virglrenderer/PKGBUILD:26]: this seems to be the connection between the patch an upstream

File: mingw-w64-virglrenderer/PKGBUILD (hunk 1 above)
Comment (heljkon) [reply mingw-w64-virglrenderer/PKGBUILD:15]: I added cmake to enable ucrt64 and clang64/32 builds.

libepoxy was not correctly detemined locally:

```
==> Starting build()...
The Meson build system
Version: 0.60.1
Source dir: C:/src/msys2/pkg/mingw-w64-virglrenderer/src/virglrenderer-virglrenderer-0.9.1
Build dir: C:/src/msys2/pkg/mingw-w64-virglrenderer/src/virglrenderer-virglrenderer-0.9.1/build
Build type: native build
Project name: virglrenderer
Project version: 0.9.1
C compiler for the host machine: clang (clang 13.0.0 "clang version 13.0.0")
C linker for the host machine: clang ld.lld 13.0.0
Host machine cpu family: x86_64
Host machine cpu: x86_64
Compiler for C supports arguments -Werror=implicit-function-declaration: YES
Compiler for C supports arguments -Werror=missing-prototypes: YES
Compiler for C supports arguments -Wmissing-prototypes: YES
Compiler for C supports arguments -Werror=int-to-pointer-cast: YES
Compiler for C supports arguments -Wno-overlength-strings: YES
Compiler for C supports arguments -fvisibility=hidden: YES
Program python3 found: YES (C:\msys64\clang64\bin/python3.EXE)
Run-time dependency threads found: YES
Found pkg-config: C:\msys64\usr\bin/pkg-config.EXE (1.8.0)
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency epoxy found: NO (tried pkgconfig)

meson.build:78:0: ERROR: Dependency "epoxy" not found, tried pkgconfig

A full log can be found at C:/src/msys2/pkg/mingw-w64-virglrenderer/src/virglrenderer-virglrenderer-0.9.1/build/meson-logs/meson-log.txt
==> ERROR: A failure occurred in build().
    Aborting...
```

Comment (lazka) [reply mingw-w64-virglrenderer/PKGBUILD]: I guess you didn't pass `-s` to makepkg so the deps were missing. It worked fine in CI.

Comment (heljkon) [reply mingw-w64-virglrenderer/PKGBUILD]: I did. Usually I build locally with

```
export MINGW_INSTALLS=mingw64
export MINGW_ARCH=mingw64
makepkg-mingw --skippgpcheck --noconfirm -sCLf mingw-w64-virglrenderer
```

Is makepkg-mingw not correct?

**Pull request merged**